tests/utils.py

```diff
@@ -56,3 +60,125 @@ def save_value(self, value, file):
 def set_seed(seed):
     random.seed(seed)
     np.random.seed(seed)
+
+
+def get_image_only_transforms(
+    custom_arguments: typing.Optional[typing.Dict[typing.Type[albumentations.ImageOnlyTransform], dict]] = None,
+    except_augmentations: typing.Optional[typing.Set[typing.Type[albumentations.ImageOnlyTransform]]] = None,
+) -> typing.List[typing.Tuple[typing.Type[albumentations.ImageOnlyTransform], dict]]:
+    custom_arguments = custom_arguments or {}
+    except_augmentations = except_augmentations or set()
+
+    result = []
+
+    for name, cls in inspect.getmembers(albumentations):
+        if not inspect.isclass(cls) or not issubclass(cls, albumentations.BasicTransform):
+            continue
+
+        if "DeprecationWarning" in inspect.getsource(cls) or "FutureWarning" in inspect.getsource(cls):
+            continue
+
+        if (
+            not issubclass(cls, albumentations.ImageOnlyTransform)
+            or cls == albumentations.ImageOnlyTransform
+            or cls in except_augmentations
+        ):
+            continue
+
+        try:
+            if issubclass(cls, albumentations.BasicIAATransform):
+                continue
+        except AttributeError:
+            pass
+
+        result.append((cls, custom_arguments.get(cls, {})))
+
+    return result
+
+
+def get_dual_transforms(
+    custom_arguments: typing.Optional[typing.Dict[typing.Type[albumentations.DualTransform], dict]] = None,
+    except_augmentations: typing.Optional[typing.Set[typing.Type[albumentations.DualTransform]]] = None,
+) -> typing.List[typing.Tuple[typing.Type[albumentations.DualTransform], dict]]:
+    custom_arguments = custom_arguments or {}
+    except_augmentations = except_augmentations or set()
+
+    result = []
+
+    for name, cls in inspect.getmembers(albumentations):
+        if not inspect.isclass(cls) or not issubclass(cls, albumentations.BasicTransform):
+            continue
+
+        if "DeprecationWarning" in inspect.getsource(cls) or "FutureWarning" in inspect.getsource(cls):
+            continue
+
+        if (
+            not issubclass(cls, albumentations.DualTransform)
+            or cls == albumentations.DualTransform
+            or cls in except_augmentations
+        ):
+            continue
+
+        try:
+            if issubclass(cls, albumentations.BasicIAATransform):
+                continue
+        except AttributeError:
+            pass
+
+        result.append((cls, custom_arguments.get(cls, {})))
+
+    return result
+
+
+def get_transforms(
```

Comment:
It looks like those three functions differ only on filtering logic for augmentation classes. Maybe it is better to implement a single function that will take an additional argument with a filter function, e.g.:

```
def get_transforms(filter_fn, custom_arguments, except_augmentations):
    ....
    if not filter_fn(cls) or cls in except_augmentations:
        continue
```

And a filter function will look something like this:

```
def filter_dual_transforms(cls):
    return issubclass(cls, albumentations.DualTransform) and not cls == albumentations.DualTransform
```

Reply:
Done